node/opr.go

```diff
@@ -1,33 +1,37 @@
 package node
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/Factom-Asset-Tokens/factom"
 	"github.com/pegnet/pegnet/modules/grader"
+	log "github.com/sirupsen/logrus"
 )
 
-func (d *Pegnetd) Grade(block *factom.EBlock) (grader.GradedBlock, error) {
+func (d *Pegnetd) Grade(ctx context.Context, block *factom.EBlock) (grader.GradedBlock, error) {
 	if block == nil {
 		// TODO: Handle the case where there is no opr block.
 		// 		Must delay conversions if this happens
 		return nil, nil
 	}
 
-	if *block.ChainID != d.Tracking["opr"] {
+	if *block.ChainID != OPRChain {
 		return nil, fmt.Errorf("trying to grade a non-opr chain")
 	}
 
 	ver := uint8(1)
-	// TODO: Handle other than mainnet?
-	if block.Height > 210330 {
+	if block.Height >= 210330 {
 		ver = 2
 	}
 
 	var prevWinners []string = nil
-	prev := d.Pegnet.FetchPreviousBlock()
-	if prev != nil {
-		prevWinners = prev.WinnersShortHashes()
+	prev, err := d.Pegnet.SelectGrade(ctx, block.Height-1)
+	// assume that error means it's below genesis for now
+	if err != nil {
+		log.WithError(err).Debug("failed to get previous winners")
```

Comment:
We can check the error for `Notfound`, otherwise, actually return an error.

This will fork a node later if we have a db issue.

Reply:
done